### PR TITLE
Adds MinikubeExtension#getDockerEnv utility function.

### LINKED ISSUE #20: Add MinikubeExtension#getDockerEnv to return the Minikube Docker environment.
Add utility function on the `MinikubeExtension`:

`getDockerEnv` -> returns a map from Docker environment variable names to their values

The Docker environment variables are retrieved via the command:

`minikube docker-env --shell=none`

Which gives for example https://github.com/kubernetes/minikube/pull/1792

These variables can then be used to configure a Docker build to build on the Minikube Docker daemon. The user will use some existing Docker build plugin to perform their Docker build.

### PR DESCRIPTION
**Pull request by coollog**
Fixes #20 .

### CODE REVIEW

File: minikube-gradle-plugin/README.md [hunk 1]
```diff
@@ -40,3 +40,19 @@ task minikubeCustom(type: com.google.cloud.tools.minikube.MinikubeTask) {
...
 ```
+
+The `minikube` extension also provides a method to obtain the environment variables necessary for working with minikube's docker daemon:
+(See [`docker-machine env` reference](https://docs.docker.com/machine/reference/env/))
+
+```
+def minikubeDockerEnv = minikube.getDockerEnv()
+```
+
+`minikubeDockerEnv` would be a map containing these fields:
```
Comment (loosebazooka) [inline minikube-gradle-plugin/README.md]: would be a map -> is a map ?

File: minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java [hunk 2]
```diff
@@ -16,16 +16,42 @@
...
+import org.gradle.api.logging.Logger;
 import org.gradle.api.provider.PropertyState;
 
 /** Minikube configuration extension. */
 public class MinikubeExtension {
   private final PropertyState<String> minikube;
+  private final Logger logger;
+
+  // @VisibleForTesting
+  class CommandExecutorFactory {
```
Comment (loosebazooka) [inline minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java]: Is this class useful outside of minikubeExtension, like in the tasks?

File: minikube-gradle-plugin/README.md (hunk 1 above)
Comment (coollog) [reply minikube-gradle-plugin/README.md]: Done

File: minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java (hunk 2 above)
Comment (coollog) [reply minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java]: We could have the `CommandExecutorFactory` be a class until `util`, and `MinikubeTask` could use that instead of calling the `CommandExecutor` constructor.

Comment (loosebazooka) [reply minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java]: Yeah, I mean, I realize it's the clean way to get stuff testable. I was just wondering. Otherwise the extension class itself becomes hard to follow and we really just want to limit it to configuration as much as we can.

Comment (coollog) [reply minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java]: I pushed a commit that moves it to its own class file. Another benefit is that future classes that need to mock the `CommandExecutor` can use the `CommandExecutorFactory` as well.

File: minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java [hunk 3]
```diff
@@ -16,16 +16,32 @@
...
+import java.util.Map;
 import org.gradle.api.Project;
 import org.gradle.api.provider.PropertyState;
 
 /** Minikube configuration extension. */
 public class MinikubeExtension {
   private final PropertyState<String> minikube;
 
+  // @VisibleForTesting
+  MinikubeExtension setCommandExecutorFactory(CommandExecutorFactory commandExecutorFactory) {
```
Comment (loosebazooka) [inline minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java]: Oops, I think you can pass this through the constructor instead of here.

Comment (loosebazooka) [reply minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java]:
```
public MinikubeExtension(Project project, CommandExecutorFactory factory)
```

and then in MinikubePlugin.java when you create the extension, you just pass in a command executor factory, and that same factory can also be set on the Task in MinikubePlugin.java

Comment (coollog) [reply minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java]: So to add it to the MinikubeTask, we would need a MinikubeTask#setCommandExecutorFactory. Is this what we want?

Comment (coollog) [reply minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java]: We would need to add a MinikubeTask#setCommandExecutorFactory method to pass the plugin's command executor factory into MinikubeTask. Is this what we want?

**Comment by coollog** [comment]
asdf

**Comment by coollog** [comment]
asdf

**Comment by coollog** [comment]
Moved the CommandExecutorFactory to be a common one created in the MinikubePlugin. However, one problem with this approach is that the MinikubeTask could be in a state where it has a null commandExecutorFactory.

File: minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubePlugin.java [hunk 4]
```diff
@@ -49,23 +51,28 @@ private void configureMinikubeTaskAdditionCallback() {
             });
```
Comment (loosebazooka) [inline minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubePlugin.java]: I think you can add commandExecutorFactory in here, so it's applied to our and custom user tasks?

Comment (coollog) [reply minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubePlugin.java]: Done

File: minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java [hunk 5]
```diff
@@ -16,16 +16,30 @@
...
+  private final CommandExecutorFactory commandExecutorFactory;
+
+  public MinikubeExtension(Project project, CommandExecutorFactory commandExecutorFactory) {
     minikube = project.property(String.class);
     setMinikube("minikube");
+
+    this.commandExecutorFactory = commandExecutorFactory;
+  }
+
+  public CommandExecutorFactory getCommandExecutorFactory() {
```
Comment (loosebazooka) [inline minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java]: Do we need this getter?

Comment (coollog) [reply minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java]: Nope, I took it out. This was leftover from another attempt.

**Comment by loosebazooka** [comment]
:ship: ship it
